x-pack/docs/en/security/authentication/remote-clusters-privileges.asciidoc

```diff
@@ -0,0 +1,225 @@
+{
+  "cluster": [
+    "read_ccr"
+  ],
+  "indices": [
+    {
+      "names": [
+        "leader-index-name"
+      ],
+      "privileges": [
+        "monitor",
+        "read"
+      ]
+    }
+  ]
+}
+----
+
+[discrete]
+===== Local cluster
+On the local cluster that contains the follower index, the `ccr_user` requires
+the `manage_ccr` cluster privilege, and `monitor`, `read`, `write`, and
+`manage_follow_index` privileges on the follower index.
+
+NOTE: If you are managing
+<<remote-clusters-connect,connecting to the remote cluster>> using
+the cluster update settings API, you will also need a user with the `all`
+cluster privilege.
+
+[source,console]
+----
+PUT /_security/role/ccr_user
+{
+  "cluster": [
+    "manage_ccr"
+  ],
+  "indices": [
+    {
+      "names": [
+        "follower-index-name"
+      ],
+      "privileges": [
+        "monitor",
+        "read",
+        "write",
+        "manage_follow_index"
+      ]
+    }
+  ]
+}
+----
+// TEST[setup:admin_role]
+
+After creating the `ccr_user` role on each cluster,
+<<security-api-put-role-mapping,map the role>> to each user that requires {ccr}
+capabilities. For example, the following request assigns the
+`ccr_user` role to a user named `my_admin_role`:
+
+[source,console]
+----
+PUT /_security/role_mapping/mapping1
+{
+  "roles": [ "ccr_user"],
+  "enabled": true, 
+  "rules": {
+    "field" : { "username" : "my_admin_role" }
+  }
+}
+----
+// TEST[continued]
+
+//end::configure-ccr-privileges[]
+
+You can then <<ccr-getting-started-tutorial,configure {ccr}>> to replicate your data
+across datacenters. 
+
+[[remote-clusters-privileges-ccs]]
+==== Configure privileges for {ccs}
+The {ccs} user requires different cluster and index privileges on the remote
+cluster and local cluster. If necessary,
+<<security-api-put-user,create a user>> that you want to assign the {ccs} role
+to.
+
+[discrete]
+===== Remote cluster
+On the remote cluster, the `ccs_user` requires the `read` and
+`read_cross_cluster` privileges for the target indices.
+
+NOTE: If requests will be issued <<run-as-privilege,on behalf of other users>>,
+then the the authenticating user must have the `run_as` privilege on the remote 
+cluster.
+
+[source,console]
+----
+PUT /_security/role/ccs_user
+{
+  "indices": [
+    {
+      "names": [
+        "target-indices"
+      ],
+      "privileges": [
+        "read",
+        "read_cross_cluster"
+      ]
+    }
+  ]
+}
+----
+
+[discrete]
+===== Local cluster
+On the local cluster, which is the cluster used to initiate cross cluster
+search, a user only needs the `ccs_user` role. The role privileges for this user
+can be empty.
+
+[source,console]
+----
+PUT /_security/role/ccs_user
+{}
+----
+// TEST[setup:admin_role]
+
+After creating the `ccs_user` role on each cluster,
+<<security-api-put-role-mapping,map the role>> to each user that requires {ccs}
+capabilities. For example, the following request assigns the `ccs_user` role to
+a user named `my_admin_role`:
```

Comment:
I am not sure why we choose to use role mapping to explain assigning the `ccr_user` role to a user. Is it simpler to just have a native user for examples by using the Create User API?

Also I find it confusing to name a "role" as `xxx_user`. I am aware that many reserved roles are named like this. But I think it helps readers if we avoid it.

Reply:
@ywangd, even if we used a native user, the role still needs to be applied to that user, correct? Or are you suggesting that we just provide an example that uses the [create user API](https://www.elastic.co/guide/en/elasticsearch/reference/master/security-api-put-user.html) to create the CCR/CCS user and assign the role in a single request? Something like:
```
POST /_security/user/cross-cluster
{
  "password" : "l0ng-r4nd0m-p@ssw0rd",
  "roles" : [ "ccr_user" ]
}
```

Reply:
Yes. I feel it is easier to follow to use native users in the example because you directly assign the necessary role to the user in the request. With role mapping, it is still unclear how it can be applied to an end-user. Sure we can link it to the doc page for role mapping, but that's another cognitive load. Also, role mapping does not apply to all users, for example, native users do not use role mapping. You need realms with external identity service, e.g. LDAP, SAML etc, which is not necessarily the best choices for getting-started guide.